packages/components/Callout/macos/Callout.swift

```diff
@@ -79,14 +79,68 @@ class CalloutView: RCTView, CalloutWindowLifeCycleDelegate {
 	// MARK: Private methods
 
 	private func showCallout() {
+		guard !isCalloutWindowShown else {
+			return
+		}
+		
 		updateCalloutFrameToAnchor()
 		calloutWindow.makeKeyAndOrderFront(self)
+		
+		// Dismiss the Callout if the window is no longer active.
+		NotificationCenter.default.addObserver(self, selector: #selector(dismissCallout), name: NSApplication.didResignActiveNotification, object: nil)
+		
+		mouseEventMonitor.addLocalMonitorForEvents(matching: .leftMouseUp, handler: { [weak self] (event) -> NSEvent? in
+			guard let clickedWindow = event.window else {
+   				return event
+   			}
+   
+   			var shouldDismissCallout = true
+   
+   			// Did the click happened in our own window?
+   			if (clickedWindow.isEqual(to: self) ) {
```

Comment:
hmmm if all we are doing is setting shoulddismisscallout maybe we should have OR operators in one if statements? amI missing something?

Reply:
I think I did it this way for readability. It's easier for me to read these `else if`'s as cases, and easier for me to add more cases in the future (for say, L2 submenus which are currently not supported..)